src/helpers/fallbackauthdialog.ts

```diff
@@ -65,3 +97,62 @@ function handleResponse(response) {
     Office.context.ui.messageParent(JSON.stringify({ status: "success", result: response.accessToken }));
   }
 }
+
+export async function dialogFallback(callback) {
+  // Attempt to acquire token silently if user is already signed in.
+  if (homeAccountId !== null) {
+    const result = await publicClientApp.acquireTokenSilent(loginRequest);
+    if (result !== null && result.accessToken !== null) {
+      const response = await callGetUserData(result.accessToken);
+      callbackFunction(response);
+    }
+  } else {
+    callbackFunction = callback;
+
+    // We fall back to Dialog API for any error.
+    const url = "/fallbackauthdialog.html";
+    showLoginPopup(url);
+  }
+}
+
+// This handler responds to the success or failure message that the pop-up dialog receives from the identity provider
+// and access token provider.
+async function processMessage(arg) {
+  console.log("Message received in processMessage: " + JSON.stringify(arg));
+  let messageFromDialog = JSON.parse(arg.message);
+
+  if (messageFromDialog.status === "success") {
+    // We now have a valid access token.
+    loginDialog.close();
+
+    // Configure MSAL to use the signed-in account as the active account for future requests.
+    homeAccountId = messageFromDialog.accountId; // Track the account id for future silent token requests.
```

Comment:
The accountId is not being sent from the fallback dialog, so this will never be received.

Reply:
I think fixing the other line will correct this right?

Reply:
It should, yes.